graph/pull.go

```diff
@@ -27,10 +18,28 @@ type ImagePullConfig struct {
 	OutStream   io.Writer
 }
 
+type Puller interface {
+	// Pull tries to pull `canonicalImageName` with tag `tag`, from `endpoint` with the config specified by `imagePullConfig`.
+	// `name` is the image's name on the server-side
+	// `sf` is an object that allows to stream back to the client (to be removed)
+	// Pull returns an error if any, as well as a boolean that determines whether to retry Pull on the next configured endpoint.
+	Pull(canonicalImageName, name, tag string, endpoint registry.APIEndpoint, imagePullConfig *ImagePullConfig, sf *streamformatter.StreamFormatter) (fallback bool, err error)
```

Comment:
I see. Can we enumerate these problems in a TODO below the interface definition?